docs/DeveloperGuide.md

```diff
@@ -8,6 +8,283 @@
+by using the `save(1)` command, which saves the first news article on the list.
+
+Step 4: When the user is done saving the desired news articles, he is able to go back to the main function by using the
+`back()` command.
+
+The flow can be seen from the sequence diagram below.
+<img src="UML Diagrams/dailyFunctionSequence.png">
+
+### Source Function
+
+
+The `sourceNews` function in the `NewsOnTheGo` class 
+is used to retrieve the source of a news article. 
+The function takes in a string and a list of 
+`NewsArticle` objects. The string is split into an 
+array and the second element (index 1) is parsed as 
+an integer. This integer is used as an index to 
+retrieve a `NewsArticle` from the list, and the 
+source of the news article is then printed.
+
+Here is the code snippet for the `sourceNews` 
+function:
+
+```java
+/**
+ * Enter the news article number as stored in the array, and it will return the source of the news article.
+ */
+static void sourceNews(String line, List<NewsArticle> list) {
+    String[] split = line.split(" ");
+    int index = Integer.parseInt(split[1]) + 1;
+    System.out.println(list.get(index).getSource());
+}
+```
+
+### Filter News by Topic Feature
+
+#### Topic Function
+
+The `showTopics` function in  `NewsOnTheGo` class is used to show the list of topics linked to the current list of news 
+articles. 
+
+This mechanism makes use of the `NewsTopic` class to store each distinct News Topic as `NewsTopic` object, stored as a 
+`newsTopics` ArrayList. 
+
+The `Topic` function is complemented by the `Filter` Function which displays the list of articles related to the 
+specified topic.
+
+#### Filter Function
+The `filterNews` function in `NewsOnTheGo` class is used to show the list of articles linked to a specific topic.
+
+This mechanism makes use of the ArrayList of `relatedNewsArticles` in a `NewsTopic` object. 
+
+This feature also implements the following operations:
+- `FilterNewsCommand#save()` — Saves the list of news articles in the topic to their reading list
+- `FilterNewsCommand#get()` — gets the details of the article and displays it to the user.
+- `FilterNewsCommand#source()` — displays the source of the article to the user.
+- `FilterNewsCommand#info()` — displays the importance, reliability and bias measure of the article to the user.
+- `FilterNewsCommand#back()` — Exits the filter topic feature.
+
+Given Below is an example usage scenario and how the filter and topic mechanism behaves at each step.
+
+Step 1. The user inputs the command `TOPICS`. The `handleCommand` method will parse the input message into the command. 
+The `TOPICS` command will cause `printAllTopics` in the UI class to be called, which will display the current list of 
+topics of the news articles.
+
+The following sequence diagram shows how the topic operation works.
+<img src="UML Diagrams/topicFunctionSequence.png">
```

Comment:
![Screenshot 2024-04-04 154557](https://github.com/nus-cs2113-AY2324S2/tp/assets/110589924/d22fef98-e18e-492c-9c1a-618e45813b1c)

Shouldn't it be "opt" instead of "alt" as it only has one path